docs/core/baseapp.md

```diff
@@ -0,0 +1,329 @@
+                |   CheckState(t)(2)   |      |                      |       |                      |
+                +----------------------+      |                      |       |                      |
+CheckTx(tx3)               |                  |                      |       |                      |
+                           v                  |                      |       |                      |
+                +----------------------+      |                      |       |                      |
+                |   CheckState(t)(3)   |      |                      |       |                      |
+                +----------------------+      +----------------------+       |                      |
+DeliverTx(tx1)             |                             |                   |                      |
+                           v                             v                   |                      |
+                +----------------------+      +----------------------+       |                      |
+                |                      |      |  DeliverState(t)(1)  |       |                      |
+                |                      |      +----------------------+       |                      |
+DeliverTx(tx2)  |                      |                 |                   |                      | 
+                |                      |                 v                   |                      |
+                |                      |      +----------------------+       |                      |
+                |                      |      |  DeliverState(t)(2)  |       |                      |
+                |                      |      +----------------------+       |                      |
+DeliverTx(tx3)  |                      |                 |                   |                      |
+                |                      |                 v                   |                      |
+                |                      |      +----------------------+       |                      |
+                |                      |      |  DeliverState(t)(3)  |       |                      |
+                +----------------------+      +----------------------+       +----------------------+
+Commit()                  |                              |                               |
+                          v                              v                               v
+                +----------------------+      +----------------------+       +----------------------+
+                |  CheckState(t+1)(0)  |      | DeliverState(t+1)(0) |       |   QueryState(t+1)    |
+                +----------------------+      |                      |       |                      |
+                          .                              .                               .
+                          .                              .                               .
```

Comment:
This graphic is hard to grasp.... Maybe more explanation of what each column represents? like the left most column could be labeled "Incoming ABCI Msgs"

Reply:
updated